packages/story-editor/src/app/pageDataUrls/pageDataUrlsProvider.tsx

```diff
@@ -41,24 +43,30 @@ function PageDataUrlProvider({ children }) {
    * @param {Page} storyPage Page object.
    * @return {Function} function to cancel image generation request
    */
-  const queuePageImageGeneration = useCallback(
-    (storyPage) => {
-      const idleTaskUid = storyPage.id;
-      const idleTask = async () => {
-        const dataUrl = await storyPageToDataUrl(storyPage, {});
-        setDataUrls((state) => ({
-          ...state,
-          [storyPage.id]: dataUrl,
-        }));
-      };
+  const queuePageImageGeneration: PageDataUrlsActions['queuePageImageGeneration'] =
+    useCallback(
+      (storyPage) => {
+        const idleTaskUid: string = storyPage.id;
+        const idleTask: () => Promise<void> = async () => {
+          const dataUrl = await storyPageToDataUrl(storyPage, {});
+          setDataUrls((state) => ({
+            ...state,
+            [storyPage?.id]: dataUrl,
+          }));
+        };
 
-      const clearQueueOfPageTask = queueIdleTask([idleTaskUid, idleTask]);
-      return () => {
-        clearQueueOfPageTask();
-      };
-    },
-    [queueIdleTask]
-  );
+        const clearQueueOfPageTask = queueIdleTask([
+          idleTaskUid,
+          idleTask,
+        ]) as () => void;
```

Comment:
This casting seems unnecessary. Probably can be resolved by properly typing the whole `useIdleTaskQueue` file

Reply:
Removed -- issue was fixed with a previous commit